src/components/utils/file.js

```diff
@@ -1,16 +1,109 @@
 import { findKey } from '~utils/object'
-import { isOfMimeType } from '~utils/blob'
+import { isOfMimeType, canvasToBlob } from '~utils/blob'
 
-const defaultFileTypesAccepted = ['jpg', 'jpeg', 'png', 'pdf']
-const maxFileSizeAcceptedByAPI = 10000000 // The Onfido API only accepts files below 10 MB
-export const validateFileTypeAndSize = (
+const DEFAULT_ACCEPTED_FILE_TYPES = ['jpg', 'jpeg', 'png', 'pdf']
+const MAX_FILE_SIZE_ACCEPTED_BY_API = 10000000 // The Onfido API only accepts files below 10 MB
+const MAX_IMAGE_FILE_SIZE_ACCEPTED = 3000000
+const validateFileTypeAndSize = (
   file,
-  acceptedTypes = defaultFileTypesAccepted
+  acceptedTypes = DEFAULT_ACCEPTED_FILE_TYPES
 ) =>
   findKey(
     {
       INVALID_TYPE: (file) => !isOfMimeType(acceptedTypes, file),
-      INVALID_SIZE: (file) => file.size > maxFileSizeAcceptedByAPI,
+      INVALID_IMAGE_SIZE: (file) =>
+        file.type.match(/image.*/) && file.size > MAX_IMAGE_FILE_SIZE_ACCEPTED,
+      INVALID_SIZE: (file) => file.size > MAX_FILE_SIZE_ACCEPTED_BY_API,
     },
     (checkFn) => checkFn(file)
   )
+
+const resizeImageFile = (file, onImageResize) => {
+  const reader = new FileReader()
+  reader.onload = (readerEvent) => {
+    const image = new Image()
+    image.onload = () => {
+      const resizeTo = getDimensionsToResizeTo(image)
+      const resizedWidth = resizeTo.width
+      const resizedHeight = resizeTo.height
+      const tempCanvas = document.createElement('canvas')
+      tempCanvas.width = resizedWidth
+      tempCanvas.height = resizedHeight
+      tempCanvas
+        .getContext('2d')
+        .drawImage(image, 0, 0, resizedWidth, resizedHeight)
+      const imgDiff = {
+        resizedFrom: {
+          width: image.width,
+          height: image.height,
+        },
+        resizedTo: {
+          width: resizedWidth,
+          height: resizedHeight,
+        },
+      }
+      return canvasToBlob(
+        tempCanvas,
+        (blob) => onImageResize({ resizedImage: blob, imgDiff }),
+        'image/png'
+      )
+    }
+    image.src = readerEvent.target.result
+  }
+  reader.readAsDataURL(file)
+}
+
+export const getDimensionsToResizeTo = (image) => {
+  // 1440px because we want to conservatively resize for Web SDK
+  // compared to mobile SDKs' 720p (1280×720px) as their UI always has a frame
+  const MAX_SIZE_IN_PIXEL = 1440
+  const ORIGNAL_WIDTH = image.width
+  let resizedWidth = ORIGNAL_WIDTH
+  const widthOnePercent = ORIGNAL_WIDTH / 100
+  const ORIGINAL_HEIGHT = image.height
+  let resizedHeight = ORIGINAL_HEIGHT
+  const heightOnePercent = ORIGINAL_HEIGHT / 100
+  let imageCurrentPercent
+  if (ORIGNAL_WIDTH > ORIGINAL_HEIGHT && ORIGNAL_WIDTH > MAX_SIZE_IN_PIXEL) {
```

Comment:
weird way you did this. It seems correct but is hard to read.

I would probably do it with ratio.

```
ratio = w / h

if ratio > 1: 
  //landscape
  new_width =  MAX_SIZE_IN_PIXEL
  new_height = height * new_width / width
else: // portrait
  new_height = MAX_SIZE_IN_PIXEL
  new_width = width * new_heigh / hiehgt
```

i think that way of coding makes all much easier to understand

Reply:
Went with resizing by percentage because my initial attempt with ratio resulted in squashed images. Your pseudocode's clued me in to where I went wrong 1st time. 😅